presto-main/src/main/java/com/facebook/presto/execution/SqlQueryQueueManager.java

```diff
@@ -51,20 +52,20 @@ public SqlQueryQueueManager(List<QueryQueueRule> rules, MBeanExporter mbeanExpor
     }
 
     @Override
-    public boolean submit(Statement statement, QueryExecution queryExecution, Executor executor)
+    public void submit(Statement statement, QueryExecution queryExecution, Executor executor)
     {
         List<QueryQueue> queues = selectQueues(statement, queryExecution.getSession(), executor);
 
         for (QueryQueue queue : queues) {
             if (!queue.reserve(queryExecution)) {
                 // Reject query if we couldn't acquire a permit to enter the queue.
                 // The permits will be released when this query fails.
-                return false;
+                queryExecution.fail(new PrestoException(QUERY_QUEUE_FULL, "Too many queued queries!"));
```

Comment:
We don't  have the name available here, but all this code will be deleted once we switch to the resource groups by default